cupy/_math/misc.py

```diff
@@ -388,12 +384,22 @@ def nan_to_num(x, copy=True, nan=0.0, posinf=None, neginf=None):
     .. seealso:: :func:`numpy.nan_to_num`
 
     """
-    dtype = x.dtype
-    nan = _check_nan_inf(nan, dtype)
-    posinf = _check_nan_inf(posinf, dtype, False)
-    neginf = _check_nan_inf(neginf, dtype, True)
-    out = None if copy else x
-    return _nan_to_num(x, nan, posinf, neginf, out=out)
+    if isinstance(x, cupy.ndarray):
+        # Early return for bool and integers as nan, posinf and neginf take no
+        # effect on them.
+        if x.dtype.kind in 'biu':
+            return x.copy() if copy else x
+
+    posinf_supplied = posinf is not None
+    posinf = posinf if posinf is not None else 0
+    neginf_supplied = neginf is not None
+    neginf = neginf if neginf is not None else 0
+    if isinstance(x, cupy.ndarray):
+        out = cupy.empty_like(x) if copy else x
+    else:
+        out = cupy.full((), x)
```

Comment:
If `out` is computed earlier, we can use `out.dtype`, `cupy.finfo(dtype).min`, and etc.

Reply:
All right! Let me make the change based on the original code, force-pushing.